File: kt/src/main/kotlin/com/github/scglwsj/parking/boy/OderParkingBoyRule.kt
package com.github.scglwsj.parking.boy

import com.github.scglwsj.parking.lot.ParkingLot

class OderParkingBoyRule : ParkingBoyRule {
    override fun findOneValidParkingLot(parkingLots: List<ParkingLot>): ParkingLot? =
        parkingLots.firstOrNull { it.hasAvailableSpaces }
}

File: cs/Parking/Boy/OderParkingBoyRule.cs
using System.Collections.Generic;
using System.Linq;

namespace Parking
{
    public class OderParkingBoyRule : IParkingBoyRule
    {
        public ParkingLot FindOneValidParkingLot(IReadOnlyList<ParkingLot> parkingLots) =>
            parkingLots.FirstOrDefault(parkingLot => parkingLot.HasAvailableSpaces);
    }
}
